### ISSUE TITLE
Include the file name when reporting Dhall errors

### ISSUE THREAD

**Issue by hdgarrood** [spacchetti/spago#222]
For example, if I modify packages.dhall so that there is a syntax error:

```
$ spago build
spago: 
Error: Invalid input

2:9:
  |
2 | Welcome to your new Dhall package-set!
  |         ^
unexpected 't'
expecting ':', '=', or whitespace
```

It would be helpful if the error message included the file name "packages.dhall". Likewise for errors involving spago.dhall.

**Comment by f-f** [comment]
@hdgarrood wow, great catch.

I'm not sure where the problem is - since the `spago.dhall` imports `packages.dhall`, the error happens on the Dhall side of things (it should be [this line](https://github.com/spacchetti/spago/blob/master/app/Spago/Config.hs#L50)), but I'm pretty sure there's an option to get nicer error reporting with filenames (i.e. I think the fix could just be replacing `inputExpr` with `inputFile` on that line)

**Comment by f-f** [comment]
Fix up in #241
